src/util.js

```diff
@@ -0,0 +1,226 @@
+  if (node instanceof Uint8Array || typeof node === 'string') {
+    node = { Data: node }
+  }
+
+  if (typeof node !== 'object' || Array.isArray(node)) {
+    throw new TypeError('Invalid DAG-PB form')
+  }
+
+  /** @type {PBNode} */
+  const pbn = {}
+
+  if (node.Data !== undefined) {
+    if (typeof node.Data === 'string') {
+      pbn.Data = textEncoder.encode(node.Data)
+    } else if (node.Data instanceof Uint8Array) {
+      pbn.Data = node.Data
+    } else {
+      throw new TypeError('Invalid DAG-PB form')
+    }
+  }
+
+  if (node.Links !== undefined) {
+    if (Array.isArray(node.Links)) {
+      pbn.Links = node.Links.map(asLink)
+      pbn.Links.sort(linkComparator)
+    } else {
+      throw new TypeError('Invalid DAG-PB form')
+    }
+  } else {
+    pbn.Links = []
+  }
+
+  return pbn
+}
+
+/**
+ * @param {PBNode} node
+ */
+export function validate (node) {
+  /*
+  type PBLink struct {
+    Hash optional Link
+    Name optional String
+    Tsize optional Int
+  }
+
+  type PBNode struct {
+    Links [PBLink]
+    Data optional Bytes
+  }
+  */
+  if (!node || typeof node !== 'object' || Array.isArray(node)) {
+    throw new TypeError('Invalid DAG-PB form')
+  }
+
+  if (!hasOnlyProperties(node, pbNodeProperties)) {
+    throw new TypeError('Invalid DAG-PB form (extraneous properties)')
+  }
+
+  if (node.Data !== undefined && !(node.Data instanceof Uint8Array)) {
+    throw new TypeError('Invalid DAG-PB form (Data must be a Uint8Array)')
+  }
+
+  if (!Array.isArray(node.Links)) {
+    throw new TypeError('Invalid DAG-PB form (Links must be an array)')
+  }
+
+  for (let i = 0; i < node.Links.length; i++) {
+    const link = node.Links[i]
+    if (!link || typeof link !== 'object' || Array.isArray(link)) {
+      throw new TypeError('Invalid DAG-PB form (bad link object)')
+    }
+
+    if (!hasOnlyProperties(link, pbLinkProperties)) {
+      throw new TypeError('Invalid DAG-PB form (extraneous properties on link object)')
+    }
+
+    if (!link.Hash) {
+      throw new TypeError('Invalid DAG-PB form (link must have a Hash)')
+    }
+
+    // @ts-ignore private property for TS
+    if (link.Hash.asCID !== link.Hash) {
+      throw new TypeError('Invalid DAG-PB form (link Hash must be a CID)')
+    }
+
+    if (link.Name !== undefined && typeof link.Name !== 'string') {
+      throw new TypeError('Invalid DAG-PB form (link Name must be a string)')
+    }
+
+    if (link.Tsize !== undefined && (typeof link.Tsize !== 'number' || link.Tsize % 1 !== 0)) {
+      throw new TypeError('Invalid DAG-PB form (link Tsize must be an integer)')
+    }
+
+    if (i > 0 && linkComparator(link, node.Links[i - 1]) === -1) {
+      throw new TypeError('Invalid DAG-PB form (links must be sorted by Name bytes)')
+    }
+  }
+}
+
+/**
+ * @param {Uint8Array} data
+ * @param {PBLink[]} [links=[]]
+ * @returns {PBNode}
+ */
+export function createNode (data, links = []) {
+  return prepare({ Data: data, Links: links })
```

Comment:
I literary meant this:

```suggestion
return { Data: data, Links: links }
```

Reply:
The reason for the `prepare()` is simply to ensure the data types and shape are correct, do some basic coercion where possible and to throw an error if this won't pass `encode()`. It's essentially doing `{ Data: data, Links: links}` but with checking too so I don't have to duplicate that logic. So the helpers are not just giving you correctly shaped objects but are checking the values you pass it too.